### PR TITLE
Validate aud using client id.

### PR DESCRIPTION
**Pull request by magic-ravi**
## Changed:
* Validate aud using client ID
* Fetch client ID from secret key when initializing client.

## Tested:
```
	api, _ := New("SECRET KEY", magic.NewDefaultClient())
	tk, _ := token.NewToken("DID TOKEN")
	validate_err := tk.Validate(api.ClientInfo.ClientId)
	assert.Equal(t, validate_err, nil)
	validate_err2 := tk.Validate("diffClientId")
	assert.NotEqual(t, validate_err2, nil)
```

### CODE REVIEW

File: token/did_test.go [hunk 1]
```diff
@@ -43,5 +43,5 @@ func TestDIDTokenDecode(t *testing.T) {
 	assert.Equal(t, proof, token.GetProof())
 	assert.Equal(t, issuer, token.GetIssuer())
 
-	assert.NoError(t, token.Validate(), "token is not valid")
+	assert.NoError(t, token.Validate("did:magic:731848cc-084e-41ff-bbdf-7f103817ea6b"), "token is not valid")
```
[the remark this answers is not in the document]
Comment (magic-ravi) [reply token/did_test.go:46]: It's to match the generated DID token (instead of re-generating the DID token manually)

**Review by justinnout — COMMENTED** [review]
Code looks good! Just a few comments

File: token/did_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (justinnout) [reply token/did_test.go:46]: I was confused by this too. Imo `magicClientId` is already overloaded, is it an Object ID (encoded | decoded) or is it this aud value? Is there another way we can represent this value?

File: cmd/magic-cli/main.go [hunk 2]
```diff
@@ -22,7 +22,7 @@ func main() {
 			{
 				Name:    "token",
 				Aliases: []string{"t"},
-				Usage:   "magic-cli token [decode|validate] --did <DID token>",
+				Usage:   "magic-cli token [decode|validate] --did <DID token> [--clientId <Magic Client ID>]",
```
[the remark this answers is not in the document]
Comment (magic-ravi) [reply cmd/magic-cli/main.go:25]: The format more closely matches `izg8mJKg4kEPngLuXhLUQgsFRehXGbfoUXpBhHiDq6U=`

File: client/api.go [hunk 3]
```diff
@@ -7,12 +7,41 @@ import (
...
-		User: user.NewUserClient(secret, client),
+		User:       user.NewUserClient(secret, clientInfo.ClientId, client),
+		ClientInfo: *clientInfo,
+	}, nil
+}
+
+func getClientInfo(secret string, client *resty.Client) (*magic.ClientInfo, error) {
+	meta := new(magic.ClientInfo)
+	respData := new(magic.Response)
+	respData.Data = meta
```
[the remark this answers is not in the document]
Comment (magic-ravi) [reply client/api.go:33]: Copied this from our existing HTTP request code. Will fix it in a minor patch.

File: token/did_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (magic-ravi) [reply token/did_test.go:46]: It's an encoded ObjectID, which is what is contained in the DID tokens we generate today (in the `aud` field).

**Comment by magic-ravi** [comment]
Will update in a patch version to include clientID as an optional param and to re-generate the DID token used for our unit tests.

File: client/api.go [hunk 4]
```diff
@@ -7,12 +7,41 @@ import (
...
+	User       magic.User
+	ClientInfo magic.ClientInfo
 }
 
-func New(secret string, client *resty.Client) *API {
+func New(secret string, client *resty.Client) (*API, error) {
+	clientInfo, err := getClientInfo(secret, client)
+	if err != nil {
+		return nil, err
+	}
```
[the remark this answers is not in the document]
Comment (magic-ravi) [reply client/api.go:23]: Will do. I'm going to add a separate constructor ending in `withClientId` as a follow-up.

**Pull request merged**